src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,80 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todos: Todo[];
+  select: number;
+  setSelect: (select: number) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todos, select, setSelect }) => {
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  const selectedTodo = todos.find(todo => todo.id === select);
+  const foundedUserId = selectedTodo?.userId;
+
+  useEffect(() => {
+    setLoading(true);
+
+    if (foundedUserId) {
+      getUser(foundedUserId)
+        .then(userFromServer => setUser(userFromServer))
+        .finally(() => setLoading(false));
+    }
+  }, [foundedUserId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+      {select !== 0 && <div className="modal-background" />}
```

Comment:
The modal background is displayed when `select` is not 0. Ensure that this logic aligns with the intended behavior of your application. If `select` is initialized with 0, verify that this condition is appropriate for showing the modal background.